### PR TITLE
cli: config: show merged config with --show-origin

### LINKED ISSUE #5119: config: add support for --show-origin
Same as `git config`:
```
$ git config --list --show-origin
file:/home/efiop/.gitconfig     user.email=ruslan@iterative.ai
file:/home/efiop/.gitconfig     user.name=Ruslan Kuprieiev
file:.git/config        core.repositoryformatversion=0
file:.git/config        core.filemode=true
file:.git/config        core.bare=false
file:.git/config        core.logallrefupdates=true
file:.git/config        remote.origin.url=git@github.com:efiop/dvc
file:.git/config        remote.origin.fetch=+refs/heads/*:refs/remotes/origin/*
file:.git/config        branch.master.remote=origin
file:.git/config        branch.master.merge=refs/heads/master
file:.git/config        remote.upstream.url=git@github.com:iterative/dvc
file:.git/config        remote.upstream.fetch=+refs/heads/*:refs/remotes/upstream/*
file:.git/config        branch.fix-4050.remote=origin
file:.git/config        branch.fix-4050.merge=refs/heads/fix-4050
file:.git/config        branch.fix-4162.remote=origin
file:.git/config        branch.fix-4162.merge=refs/heads/fix-4162
```

i'm not personally sold on `file:` prefix in path (just because I don't really know why git does that, maybe there is a good reason), but the rest looks pretty good. We now have `dvc config --list`, so now we need `dvc config --list --show-origin` as the next step.

This will be realy handy to have for users to figure out where the configs are and also handy for us, developers, when debugging particular issues.

### PR DESCRIPTION
**Pull request by mrstegeman**
Builds on #5126 and #5184 by showing the full merged config when
listing or getting config options.

References #5126
References #5184
References iterative/dvc.org#2028
Fixes #5119

* [X] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [X] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

**Review by efiop — COMMENTED** [review]
Looks good! :rocket: 

Do you plan on introducing tests for this?

**Comment by mrstegeman** [comment]
The existing tests are still valid, but I've added some additional ones to cover some additional cases.

File: tests/func/test_config.py [hunk 1]
```diff
@@ -278,3 +284,13 @@ def test_config_show_origin(tmp_dir, dvc, caplog):
...
         in caplog.text
     )
+
+    caplog.clear()
+    assert main(["config", "--list", "--show-origin"]) == 0
+    assert (
+        "{}\t{}\n".format(
+            os.path.join(".dvc", "config"),
+            "remote.myremote.url=s3://bucket/path",
+        )
```
Comment (efiop) [inline tests/func/test_config.py:294]: We need a test for multiple levels in one output, just testing .dvc/config is not enough, we need to at least try it + .dvc/config.local
```suggestion
    assert (
        "{}\t{}\n".format(
            os.path.join(".dvc", "config"),
            "remote.myremote.url=s3://bucket/path",
        )
```

Also, this `test_` is pretty packed now, please consider creating separate test cases for new tests.

Comment (mrstegeman) [reply tests/func/test_config.py:306]: Done.

**Review by efiop — APPROVED** [review]
Thank you!

**Pull request merged**